internal/mock/cloudsql.go

```diff
@@ -30,45 +35,43 @@ type FakeCSQLInstance struct {
 	project string
 	region  string
 	name    string
-
-	dbVersion string
-
-	privKey *rsa.PrivateKey
+	version string
+	key     *rsa.PrivateKey
 	cert    *x509.Certificate
+	tlsCert tls.Certificate
 }
 
 // NewFakeCSQLInstance returns a CloudSQLInst object for configuring mocks.
-func NewFakeCSQLInstance(project, region, name string) (FakeCSQLInstance, error) {
+func NewFakeCSQLInstance(project, region, name string) FakeCSQLInstance {
 	// TODO: consider options for this?
-	privKey, cert, err := generateInstanceCerts()
+	key, cert, tlsCert, err := generateCerts(project, name)
```

Comment:
The differences between cert and tlscert are not clear to me

Reply:
In a later commit, I end up making Cert and Key public, which means we can build a TLS config in the mock listener when we need it. So, I'll drop the tlsCert here and build it when it's needed.